2000-10-27  Jakub Jelinek  <jakub@redhat.com>

	* expr.c (do_store_flag): If op0 has mode VOIDmode, pass operand_mode
	instead to expand_shift.

--- gcc/expr.c.jj	Fri Oct 27 13:34:59 2000
+++ gcc/expr.c	Fri Oct 27 15:54:34 2000
@@ -10285,7 +10285,7 @@ do_store_flag (exp, target, mode, only_c
       op0 = expand_expr (inner, subtarget, VOIDmode, 0);
 
       if (bitnum != 0)
-	op0 = expand_shift (RSHIFT_EXPR, GET_MODE (op0), op0,
+	op0 = expand_shift (RSHIFT_EXPR, operand_mode, op0,
 			    size_int (bitnum), subtarget, ops_unsignedp);
 
       if (GET_MODE (op0) != mode)
